integration-tests-support/google/src/main/java/org/apache/camel/quarkus/test/support/google/GoogleCloudContext.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.test.support.google;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GoogleCloudContext {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GoogleCloudContext.class);
+
+    private final ArrayList<AutoCloseable> closeables = new ArrayList<>();
+    private final Map<String, String> properties = new LinkedHashMap<>();
+    private boolean usingMockBackend;
+
+    public <C> Optional<C> getResource(Class<C> type) {
```

Comment:
It the intent of this method to do `getCloseable(Class<C> type)` ? Or maybe `Resource` means something specific in the context of google-test-support ?

Reply:
This is the code I created in my first draft of the test. It was replaced by the similar technique which is used in aws. Therefore it should be removed, Thanks for noticing!

Reply:
This method was replaced with the automatic `AutoCloseable` registration  and execution from the `GoogleCloudContext.close`. It will be removed. Thanks for noticing!

Reply:
removed. (I refactored the only one place, this method was called from)

Reply:
Ok, good we have spotted this :+1: